libraries/bio/src/utils/units-handler.ts

```diff
@@ -32,16 +28,54 @@ export class UnitsHandler {
   public static readonly RnaFastaAlphabet = new Set<string>(['A', 'C', 'G', 'U']);
 
   public static setUnitsToFastaColumn(col: DG.Column) {
+    if (col.semType !== DG.SEMTYPE.MACROMOLECULE || col.getTag(DG.TAGS.UNITS) != NOTATION.FASTA)
+      throw new Error(`The column of notation '${NOTATION.FASTA}' must be '${DG.SEMTYPE.MACROMOLECULE}'.`);
+
+    col.setTag(DG.TAGS.UNITS, NOTATION.FASTA);
+    UnitsHandler.setTags(col, 5, splitterAsFasta);
+  }
+
+  public static setUnitsToSeparatorColumn(col: DG.Column, separator?: string) {
+    if (col.semType !== DG.SEMTYPE.MACROMOLECULE || col.getTag(DG.TAGS.UNITS) != NOTATION.SEPARATOR || !separator)
+      throw new Error(`The column of notation '${NOTATION.SEPARATOR}' must be '${DG.SEMTYPE.MACROMOLECULE}'.`);
+    if (!separator)
+      throw new Error(`The column of notation '${NOTATION.SEPARATOR}' must have the separator tag.`);
+
+    col.setTag(DG.TAGS.UNITS, NOTATION.SEPARATOR);
+    col.setTag(TAGS.separator, separator);
+    UnitsHandler.setTags(col, 5, getSplitterWithSeparator(separator));
+  }
+
+  public static setUnitsToHelmColumn(col: DG.Column) {
     if (col.semType !== DG.SEMTYPE.MACROMOLECULE)
-      throw new Error('Fasta column must be MACROMOLECULE');
+      throw new Error(`The column of notation '${NOTATION.HELM}' must be '${DG.SEMTYPE.MACROMOLECULE}'`);
 
-    const stats: SeqColStats = getStats(col, 5, splitterAsFasta);
-    const aligned = stats.sameLength ? 'SEQ.MSA' : 'SEQ';
-    const alphabet = detectAlphabet(stats);
+    col.setTag(DG.TAGS.UNITS, NOTATION.HELM);
+    UnitsHandler.setTags(col, 5, splitterAsHelm);
+  }
 
-    col.setTag(DG.TAGS.UNITS, NOTATION.FASTA);
-    col.setTag(TAGS.aligned, aligned);
-    col.setTag(TAGS.alphabet, alphabet);
+  /** From detectMacromolecule */
+  public static setTags(col: DG.Column, seqMinLength: number, splitter: SplitterFunc): void {
```

Comment:
you don't use the `seqMinLength` parameter, guess you wanted to use it on line 60 instead of `5`